src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -172,15 +172,24 @@ var DOMRenderer = ReactFiberReconciler({
         namespace = root ? root.namespaceURI : getChildNamespace(null, '');
         break;
       }
-      default: {
-        const container: any = nodeType === COMMENT_NODE
-          ? rootContainerInstance.parentNode || rootContainerInstance
-          : rootContainerInstance;
+      case COMMENT_NODE: {
+        invariant(
+          rootContainerInstance.parentNode != null, // match null & undefined
```

Comment:
I don't think `parentNode` can ever be `undefined`, can it? I would expect the browser to always give us either the node or `null`.

Even if it can be `undefined`, let's just explicitly test two cases. We’ve been doing this elsewhere for explicitness too. This is also a tiny bit faster than `!= null` check because `!=` has to generate some additional special code for historical reasons.

Reply:
Sounds good - will fix.